### ISSUE TITLE
[Bug] Running pulsar-shell in docker container throws exception

### ISSUE THREAD

**Issue by mukesh154** [apache/pulsar#22440]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Read release policy

- [X] I understand that unsupported versions don't get bug fixes. I will attempt to reproduce the issue on a supported version of Pulsar client and Pulsar broker.

### Version

3.2.2

### Minimal reproduce step

**Describe the bug**

- Run below command to start pulsar into docker container:
    ```
    $ docker run -it \
    -p 6650:6650 \
    -p 8080:8080 \
    --mount source=pulsardata,target=/pulsar/data \
    --mount source=pulsarconf,target=/pulsar/conf \
    apachepulsar/pulsar:3.2.2 \
    bin/pulsar standalone
    ```
- The pulsar standalone service should start successfully. 
- exec into the docker container with default user `pulsar` using below command:
       `docker exec -it <container-id> /bin/sh
`
- Try to start the pulsar-shell with `./bin/pulsar-shell`.
- It should throw below exception:
```
   $ ./bin/pulsar-shell
        Exception in thread "main" java.nio.file.AccessDeniedException: /home/pulsar
	        at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:90)
	        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106)
	        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
	        at java.base/sun.nio.fs.UnixFileSystemProvider.createDirectory(UnixFileSystemProvider.java:397)
	        at java.base/java.nio.file.Files.createDirectory(Files.java:700)
	        at java.base/java.nio.file.Files.createAndCheckIsDirectory(Files.java:807)
	        at java.base/java.nio.file.Files.createDirectories(Files.java:793)
	        at org.apache.pulsar.shell.PulsarShell.<init>(PulsarShell.java:160)
	        at org.apache.pulsar.shell.PulsarShell.<init>(PulsarShell.java:142)
	        at org.apache.pulsar.shell.PulsarShell.main(PulsarShell.java:215)
```

- Also, if we do exec into the container with root user `docker exec -it -u root <container-id> /bin/sh`, the `pulsar-shell` is working.
- The problem is: `bin` directory is owned by `root` user not the default `pulsar` user.

### What did you expect to see?

With default `pulsar` user, the `pulsar-shell` should be opened.

### What did you see instead?

- It is throwing below exception:
```
   $ ./bin/pulsar-shell
        Exception in thread "main" java.nio.file.AccessDeniedException: /home/pulsar
	        at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:90)
	        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106)
	        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
	        at java.base/sun.nio.fs.UnixFileSystemProvider.createDirectory(UnixFileSystemProvider.java:397)
	        at java.base/java.nio.file.Files.createDirectory(Files.java:700)
	        at java.base/java.nio.file.Files.createAndCheckIsDirectory(Files.java:807)
	        at java.base/java.nio.file.Files.createDirectories(Files.java:793)
	        at org.apache.pulsar.shell.PulsarShell.<init>(PulsarShell.java:160)
	        at org.apache.pulsar.shell.PulsarShell.<init>(PulsarShell.java:142)
	        at org.apache.pulsar.shell.PulsarShell.main(PulsarShell.java:215)
```

### Anything else?

The issue is also reproducible on kubernetes environment.

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

**Comment by nicoloboschi** [comment]
@lhotari I agree with you that we should check if the directory is writable instead of throwing error. 

The directory is also used for storing cluster connection parameters when [you create a config](https://pulsar.apache.org/docs/next/administration-pulsar-shell/#connect-to-your-cluster).

I think we can do two things (maybe both)
1. Add a WARNING log saying that "new configs and history will be lost after you close the session"
2. Create an ad-hoc directory for the docker image (/pulsar/pulsar-shell) and adjust the code to look for that directory too ("user.dir"/"pulsar-shell")

**Comment by lhotari** [comment]
> I think we can do two things (maybe both)
> 
> 1. Add a WARNING log saying that "new configs and history will be lost after you close the session"

yes

> 2. Create an ad-hoc directory for the docker image (/pulsar/pulsar-shell) and adjust the code to look for that directory too ("user.dir"/"pulsar-shell")

It's better not to have custom solutions for the docker image. In the Dockerfile, the invalid home directory should be fixed. It could point to /pulsar/data, for example.

**Comment by lhotari** [comment]
I created #22446 which will resolve the issue, however it would be useful to implement the checks to pulsar-shell as well.

**Comment by lhotari** [comment]
Fixed by #22446
